symfony/plugins/orangehrmPimPlugin/Controller/SaveCustomFieldController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Pim\Controller;
+
+use OrangeHRM\Core\Controller\AbstractVueController;
+use OrangeHRM\Core\Vue\Component;
+use OrangeHRM\Core\Vue\Prop;
+use OrangeHRM\Framework\Http\Request;
+
+class SaveCustomFieldController extends AbstractVueController
+{
+    public const SCREEN_PERSONAL_DETAILS = 'Personal Details';
+    public const SCREEN_CONTACT_DETAILS = 'Contact Details';
+    public const SCREEN_EMERGENCY_CONTACTS = 'Emergency Contacts';
+    public const SCREEN_DEPENDENTS = 'Dependents';
+    public const SCREEN_IMMIGRATION = 'Immigration';
+    public const SCREEN_JOB = 'Job';
+    public const SCREEN_SALARY = 'Salary';
+    public const SCREEN_TAX_EXEMPTIONS = 'Tax Exemptions';
+    public const SCREEN_REPORT_TO = 'Report-to';
+    public const SCREEN_QUALIFICATIONS = 'Qualifications';
+    public const SCREEN_MEMBERSHIP = 'Memberships';
+
+    Const DROP_DOWN = 'Drop Down';
+    Const TEXT_NUMBER = 'Text or Number';
+
+    private array $screenList = [
+        ['id' => self::SCREEN_PERSONAL_DETAILS, 'label' => self::SCREEN_PERSONAL_DETAILS],
```

Comment:
is this the way it handles in 4.x?

Reply:
yes, the items are hardcoded in the CustomFieldForm class. The logic is same.

Reply:
I checked in 4.x,
seems labels are correct. But ids wrong
```php
class CustomField
    const SCREEN_PERSONAL_DETAILS   = 'personal';
    const SCREEN_CONTACT_DETAILS    = 'contact';
    const SCREEN_EMERGENCY_CONTACTS = 'emergency';
    const SCREEN_DEPENDENTS         = 'dependents';
    const SCREEN_IMMIGRATION        = 'immigration';
    const SCREEN_QUALIFICATIONS     = 'qualifications';
    const SCREEN_TAX_EXEMPTIONS     = 'tax';
    const SCREEN_SALARY             = 'salary';
    const SCREEN_JOB                = 'job';
    const SCREEN_REPORT_TO          = 'report-to';
    const SCREEN_MEMBERSHIP         = 'membership'; 
```

Also need to validate these keys in API

Reply:
changed :)